frontend/src/main/java/com/google/tripmeout/frontend/storage/PlaceVisitStorage.java

```diff
@@ -27,31 +31,29 @@
    * @throws a PlaceVisitNotFound exception if there is no PlaceVisitModel
    *     object in storage with the given tripId and placeId
    */
-  void removePlaceVisit(String tripId, String placeId);
+  void removePlaceVisit(String tripId, String placeId) throws PlaceVisitNotFoundException;
 
   /**
    * returns the PlaceVisitModel whose tripId and placeId match the given tripId
    * and placeId, respectively
    *
    * @param tripId the id to match PlaceVisitModel object's placeId field on
    * @param placeId the id to match PlaceVisitModel object's placeId field on
-   *
-   * @throws a PlaceVisitNotFound exception if there is no PlaceVisitModel
-   *     object in storage with the given tripId and placeId
    */
-  PlaceVisitModel getPlaceVisit(String tripId, String placeId);
+  Optional<PlaceVisitModel> getPlaceVisit(String tripId, String placeId);
 
   /**
    * updates the userMark parameter of the PlaceVisitModel object whose tripId
-   * and placeId match the given tripId and placeId, respectively
+   * and placeId match the given tripId and placeId, respectively,
+   * if PlaceVisitModel obbject in storage
+   * adds PlaceVisitModel object otherwise
+   * return true if PlaceVisitModel object with the given tripId and placeId
+   * previously exist in storage
    *
-   * @param tripId the id to match PlaceVisitModel object's tripId field on
-   * @param placeId the id to match PlaceVisitModel object's placeId field on
-   *
-   * @throws a PlaceVisitNotFound exception if there is no PlaceVisitModel
-   *     object in storage with the given tripId and placeId
+   * @param placeVisit the PlaceVisitModel object to update or add
+   * @param newStatus the new status to update placeVisit's userMark field to
    */
-  void changePlaceVisitStatus(String tripId, String placeId, String newStatus);
+  boolean updateUserMarkOrAddPlaceVisit(PlaceVisitModel placeVisit, String newStatus);
```

Comment:
nit: If this takes a `PlaceVisitModel` why not just update the store with that value?

Reply:
I'm not sure this will actually be an issue in our use but just in case other fields of placeVisit differ but the tripId and placeId are the same as a PlaceVIsitModel object in storage - I want to only update the userMark field and not the other fields

Reply:
I'm not sure this will actually be an issue in our use but just in case other fields of placeVisit differ but the tripId and placeId are the same as a PlaceVIsitModel object in storage - I want to only update the userMark field and not the other fields